pilot/pkg/networking/core/loadbalancer/loadbalancer.go

```diff
@@ -77,23 +77,23 @@ func ApplyLocalityLBSetting(
 
 	// one of Distribute or Failover settings can be applied.
 	if localityLB.GetDistribute() != nil {
-		applyLocalityWeight(locality, loadAssignment, localityLB.GetDistribute())
+		applyLocalityWeights(locality, loadAssignment, localityLB.GetDistribute())
 		// Failover needs outlier detection, otherwise Envoy will never drop down to a lower priority.
 		// Do not apply default failover when locality LB is disabled.
 	} else if enableFailover && (localityLB.Enabled == nil || localityLB.Enabled.Value) {
-		if len(localityLB.FailoverPriority) > 0 {
-			applyPriorityFailover(loadAssignment, wrappedLocalityLbEndpoints, proxyLabels, localityLB.FailoverPriority)
-			if len(localityLB.Failover) != 0 {
-				applyLocalityFailover(locality, loadAssignment, localityLB.Failover)
-			}
-			return
+		switch {
+		case len(localityLB.FailoverPriority) > 0:
```

Comment:
nit: not sure why this is better than `if/else`?

Reply:
Oh is it to reuse the `applyLocalityFailover`? I don't think that works if so -- you need to explicitly use `fallthrough`?

Reply:
I tend to if/else

Reply:
switch cases are for cases more than three

Reply:
changed. My initial idea was to reuse the call but was not able to do because of special case where failoverPriorities and failover can be set together.

BTW, our api docs need update?